plaidml/op/op.h

```diff
@@ -331,7 +331,7 @@ inline edsl::Tensor elu(const edsl::Tensor& I, double alpha) {
 class explicit_padding {
  public:
   explicit explicit_padding(const edsl::Tensor& I, const std::vector<int>& lo_pads, const std::vector<int>& hi_pads)
-      : I_(I), lo_pads_(lo_pads), hi_pads_(hi_pads), mode_(PadMode::CONSTANT), padval_(0) {}
+      : I_(I), lo_pads_(lo_pads), hi_pads_(hi_pads), mode_(PadMode::CONSTANT), padval_(edsl::Constant(0)) {}
```

Comment:
Is it nicer to say `Tensor(0)` or to say `Constant(0)`? I've been internally debating whether I should drop the `Constant()` for scalars.

Reply:
I don't have a strong opinion here, but I thought we were trying to avoid exposing the `Tensor` API based on this comment. Perhaps I am misunderstanding what you were trying to say?

> Dropped the plaidml::edsl::Tensor constructors that allow buliding of placeholders and constants. Use plaidml::edsl::Placeholder() and plaidml::edsl::Constant() instead.

Reply:
That comment was referring to Tensors/Buffers, not scalars.

Reply:
@tzerrell that is a good point. Using `Constant` for scalars would also prevent a situation where someone has to think "Oh, I can just use `Constant` to store my weights, except for scalars, where I'd have to use `Tensor`."